govcd/nsxt_segment_profiles.go

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright 2023 VMware, Inc.  All rights reserved.  Licensed under the Apache v2 License.
+ */
+
+package govcd
+
+import (
+	"net/url"
+
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+)
+
+// GetAllIpDiscoveryProfiles retrieves all IP Discovery Profiles configured in an NSX-T manager.
+// NSX-T manager ID (nsxTManagerRef.id), Org VDC ID (orgVdcId) or VDC Group ID (vdcGroupId) must be
+// supplied as a filter. Results can also be filtered by a single profile ID
```

Comment:
What happens if we don't supply the filters, does it return an error or does it silently return a 0 length slice?
Asking as I'm not sure whether we should check that the filter contains these values, if it doesn't make sense to call it without them.

Reply:
We get an error:

> error getting all pages for endpoint https://HOST/cloudapi/1.0.0/nsxTResources/segmentIpDiscoveryProfiles: error in HTTP GET request: BAD_REQUEST - [ b0f733c0-64b8-4f00-be6d-c6c08b7ff74d ] filter field value missing"}

I'd think that it could be sufficient and it actually provides a valid error.